packages/@lwc/ssr-compiler/src/transmogrify.ts

```diff
@@ -20,14 +21,18 @@ export type Visitors = Parameters<typeof traverse<Node, TransmogrificationState>
 const EMIT_IDENT = b.identifier('$$emit');
 // Rollup may rename variables to prevent shadowing. When it does, it uses the format `foo$0`, `foo$1`, etc.
 const TMPL_FN_PATTERN = /tmpl($\d+)?/;
-const GEN_MARKUP_PATTERN = /generateMarkup($\d+)?/;
+const GEN_MARKUP_OR_GEN_SLOTTED_CONTENT_PATTERN = /generate(?:Markup|SlottedContent)($\d+)?/;
 
 const isWithinFn = (pattern: RegExp, nodePath: NodePath): boolean => {
     const { node } = nodePath;
     if (!node) {
         return false;
     }
-    if (node.type === 'FunctionDeclaration' && pattern.test(node.id.name)) {
+    if (
+        (node.type === 'FunctionDeclaration' || node.type === 'FunctionExpression') &&
+        node.id &&
+        pattern.test(node.id.name)
+    ) {
```

Comment:
Function declarations are when you have the function on their own line, function expressions are when it's just passed in to another function or the value in an object etc.